engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/JobExecutionEngine.scala

```diff
@@ -35,8 +35,14 @@ final class JobExecutionEngine(
   private val backgroundJobsRef =
     new AtomicReference[Vector[RunningJob]](Vector.empty)
 
+  private val pendingCancellations =
+    new AtomicReference[Vector[(Long, RunningJob)]](Vector.empty)
+
   private val context = interpreterContext.executionService.getContext
 
+  private val pendingCancellationsExecutor =
+    context.newFixedThreadPool(1, "pending-cancellations-timer", false)
```

Comment:
`EnsoContext.newFixedThreadPool` returns [ExecutorService](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/ExecutorService.html). Can we wrap the default executor services with `SafepointAwareExecutorService` and whenever a method returning `Future` return a wrapper `SafepointAwareFuture` that would `cancel(true)` by asking `ThreadManager` first and only later emit an interrupt?